internal/kafka/internal/handlers/observatorium_proxy.go

```diff
@@ -22,30 +22,31 @@ func NewObservatoriumProxyHandler(clusterService services.ClusterService) *obser
 
 // Delete is the handler for deleting a kafka request
 func (h observatoriumProxyHandler) ValidateTokenAndExternalClusterID(w http.ResponseWriter, r *http.Request) {
+	external_id := mux.Vars(r)["cluster_external_id"]
 	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.ValidateExternalClusterId(&external_id, "external cluster id"),
+		},
 		Action: func() (i interface{}, serviceError *errors.ServiceError) {
-			external_id := mux.Vars(r)["cluster_external_id"]
-
 			ctx := r.Context()
 
 			claims, err := auth.GetClaimsFromContext(ctx)
 			if err != nil {
 				return nil, errors.Unauthenticated("unable to authenticate token provided in the request")
 			}
 
-			clientID, err := claims.GetClientId()
+			clientID, err := claims.GetClientID()
 			if err != nil {
 				return nil, errors.Unauthenticated("unable to get client ID from the token")
 			}
 
-			clusters, err := h.clusterService.FindAllClusters(services.FindClusterCriteria{})
+			clusters, err := h.clusterService.FindAllClusters(services.FindClusterCriteria{ExternalID: external_id})
```

Comment:
Here you are calling `FindAllClusters`, but you added support for the `ExternalID` in the `FindClusters` method.

Side comment: FindAllClusters and FindCluster are identical but the return value: the first one returns an array, while the second one returns the first result. I believe `FindCluster` should call `FindAllClusters` to remove duplicated code

Reply:
I switched to use FindCluster

Reply:
Mind adding the new `ExternalID` support to `FindAllClusters` too?

Reply:
sure